web/src/components/Toast.tsx

```diff
@@ -1,82 +1,133 @@
+  },
+  [ToastType.INFO]: {
+    Icon: Info,
+    color: 'text-blue-800 dark:text-blue-400 before:bg-blue-800 dark:before:bg-blue-400',
+  },
 };
 
-function Toast({
+export function Toast({
   title,
   description,
   toastAction,
   toastActionText,
   toastClose,
   toastCloseText,
   duration,
-}: Props) {
+  type = ToastType.INFO,
+}: ToastProps) {
   const { t } = useTranslation();
   const [open, setOpen] = useState(true);
   const handleToastAction = () => {
     toastAction?.();
     setOpen(false);
   };
+
   const handleToastClose = () => {
     toastClose?.();
     setOpen(false);
   };
 
+  const { Icon, color } = ToastTypeTheme[type];
+
   return (
     <>
       <ToastPrimitive.Root
+        data-testid="toast"
         open={open}
         onOpenChange={setOpen}
         duration={duration}
         type="background"
-        className="fixed top-16 z-50 m-2 inline-block max-w-max self-center rounded-lg bg-white/80 shadow backdrop-blur-sm dark:bg-gray-800/80 xs:left-1/2 xs:w-max xs:-translate-x-1/2 xs:transform"
+        className={twMerge(
+          'fixed left-1/2 top-16 z-50 inline-block flex w-max min-w-80 max-w-full -translate-x-1/2 transform rounded-lg shadow',
+          'border border-solid border-neutral-50 bg-neutral-50 dark:border-gray-700 dark:bg-gray-700 dark:bg-gray-900',
+          "before:content[''] before:absolute before:block before:h-full before:w-1 before:rounded-bl-md before:rounded-tl-md",
+          color,
+          toastAction ? 'h-[52px]' : 'h-11'
+        )}
       >
-        <div className="flex items-center p-2">
-          <div className="flex flex-col">
-            <ToastPrimitive.Title className="text-md font-medium text-gray-900 dark:text-gray-100">
-              {title}
-            </ToastPrimitive.Title>
-            <ToastPrimitive.Description className="text-sm text-gray-700 dark:text-gray-400">
-              {description}
-            </ToastPrimitive.Description>
+        <div className="flex w-full items-center p-2">
+          <div className="flex items-center gap-2 px-2">
+            <div className={color}>
+              <Icon size={16} data-testid="toast-icon" />
+            </div>
+            <div className="flex flex-col">
+              {title && (
+                <ToastPrimitive.Title className="text-md font-bold">
+                  {title}
+                </ToastPrimitive.Title>
+              )}
+              {description && (
+                <ToastPrimitive.Description className="text-sm font-semibold">
+                  {description}
+                </ToastPrimitive.Description>
+              )}
+            </div>
           </div>
-          {toastAction && (
-            <ToastPrimitive.Action
-              altText="view now"
-              className="mx-2 flex items-center justify-center"
-              onClick={handleToastAction}
+          <div className="ml-auto flex">
+            {toastAction && (
```

Comment:
probs worth refactoring this component at some point so that `toastAction` and `toastActionText` are more tightly coupled